### PR TITLE
feat: allow setting message flags when sending

### PR DESCRIPTION
**Pull request by almeidx**
**Please describe the changes this PR makes and why it should be merged:**
- https://github.com/discord/discord-api-docs/pull/4356
- https://github.com/discord/discord-api-docs/pull/4357

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

File: packages/discord.js/src/structures/interfaces/TextBasedChannel.js [hunk 1]
```diff
@@ -72,6 +72,8 @@ class TextBasedChannel {
    * @typedef {BaseMessageOptions} MessageOptions
    * @property {ReplyOptions} [reply] The options for replying to a message
    * @property {StickerResolvable[]} [stickers=[]] Stickers to send in the message
+   * @property {MessageFlags} [flags] Which flags to set for the message.
+   * Only `SUPPRESS_EMBEDS` and `EPHEMERAL` can be set.
```
Comment (vladfrangu) [inline packages/discord.js/src/structures/interfaces/TextBasedChannel.js:76]: This is wrong. Only suppress_embeds can be set here at this time

**Review by vladfrangu — DISMISSED** [review]

**Comment by vladfrangu** [comment]
You need to document it for Interaction replies that EPHEMERAL can be used, which may be painful..

File: packages/discord.js/src/structures/MessagePayload.js [hunk 2]
```diff
@@ -148,11 +148,13 @@ class MessagePayload {
...
-    if (this.isMessage || this.isMessageManager) {
+    if (typeof this.options.flags !== 'undefined' || this.isMessage || this.isMessageManager) {
       // eslint-disable-next-line eqeqeq
       flags = this.options.flags != null ? new MessageFlags(this.options.flags).bitfield : this.target.flags?.bitfield;
-    } else if (isInteraction && this.options.ephemeral) {
-      flags = MessageFlags.FLAGS.EPHEMERAL;
+    }
+
+    if (isInteraction && this.options.ephemeral) {
+      flags |= MessageFlags.FLAGS.EPHEMERAL;
```
Comment (suneettipirneni) [inline packages/discord.js/src/structures/MessagePayload.js:157]: You can use -types `MessageFlag` here instead. I'll be removing the `FLAGS` prop from the bitfield in my other PR.

**Review by suneettipirneni — CHANGES_REQUESTED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Comment by almeidx** [comment]
Rebased

**Review by iCrawl — APPROVED** [review]

**Comment by almeidx** [comment]
Upstream pull request has been merged

**Pull request merged**